### PR TITLE
[SPARK-2458] Make failed application log visible on History Server

### PR DESCRIPTION
**Pull request by tsudukim**
Enabled HistoryServer to show incomplete applications.
We can see the log for incomplete applications by clicking the bottom link.

### CODE REVIEW

**Comment by tsudukim** [comment]
resolved conflicts!

File: core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala [hunk 1]
```diff
@@ -180,14 +176,15 @@ private[history] class FsHistoryProvider(conf: SparkConf) extends ApplicationHis
...
+              appListener.sparkUser.getOrElse(NOT_STARTED),
+              !fs.isFile(new Path(dir.getPath(), EventLoggingListener.APPLICATION_COMPLETE))))
           } catch {
             case e: Exception =>
               logInfo(s"Failed to load application log data from $dir.", e)
               None
           }
         }
-        .sortBy { info => -info.endTime }
+        .sortBy { info => (-info.endTime, -info.startTime) }
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala]: I see

File: core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala [hunk 2]
```diff
@@ -329,6 +319,17 @@ private[history] class FsHistoryProvider(conf: SparkConf) extends ApplicationHis
   /** Returns the system's mononotically increasing time. */
   private def getMonotonicTimeMs(): Long = System.nanoTime() / (1000 * 1000)
 
+  /**
+   * Return true when the application already completed.
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala]:
```
Return true when the application has completed.
```

Don't worry I'll fix the wording myself when I merge this

**Comment by andrewor14** [comment]
Alright LGTM I'm merging this into master thanks!

**Pull request closed without merging**